### DISCUSSION TITLE
Removing the Karaoke

### DISCUSSION THREAD

**Discussion by LWinterberg** [audacity/audacity#2451] [Development]
I've heard from various sides that the Karaoke tool only was intended to be a demo of a class, and that it's amazing it even is still in the code. 

So how about we throw it out?

**Comment by SteveDaulton** [comment]
It may have been written only as a demo, but it will upset those people that use it if it is removed.
What's the motivation for removing it?

**Reply by LWinterberg** [reply]
> It may have been written only as a demo, but it will upset those people that use it if it is removed.

Considering that #1738 got first reported *a decade* after its introduction and there's like 8 threads in the forum talking about it (rather than creating karaoke versions by removing the vocals) I doubt this is any significant amount of people. 

> What's the motivation for removing it?

* It's somewhat out-of-place in Audacity. A singalong Karaoke thing is more of a consumer feature than a creator feature.
* Making a karaoke using bazillions of labels is massively tedious
* It requires constant maintenance whenever under-the-hood changes happen (which will pick up in the future)
   * ... and already has caused crashes (eg #1312)

**Reply by Paul-Licameli** [reply]
> > What's the motivation for removing it?
> 
> * It's somewhat out-of-place in Audacity. A singalong Karaoke thing is more of a consumer feature than a creator feature.
> * Making a karaoke using bazillions of labels is massively tedious
> * It requires constant maintenance whenever under-the-hood changes happen (which will pick up in the future)
>   
>   * ... and already has caused crashes (eg [Bouncing Beach Ball of DEATH #1312](https://github.com/audacity/audacity/issues/1312))

I remember that crash bug.  It was an error only recently introduced at the time, not an old problem in the feature.

Leaving it in is harmless.

**Reply by SteveDaulton** [reply]
Audacity has lots of features that are only used by a small minority of users, but Audacity's user base includes a vast number of minority interests. Rather than just "throwing out" minority features, a better approach might be to make them available as optional modules (I'd guess that Paul has things to say about that).

**Reply by Paul-Licameli** [reply]
Really what's the harm of leaving this unusual feature hidden in a menu?  It's not getting in anybody's way.

Amusingly, the tedious generation of the labels might become an easy thing to do as an application of deep learning if those developments progress.  Maybe @hugofloresgarcia or a colleague will give us speech recognition label tracks.

**Reply by SteveDaulton** [reply]
> the tedious generation of the labels might become an easy thing to do as an application of deep learning if those developments progress.

Even a Nyquist plug-in could probably help (thinking of a modified "Sound Finder")

**Reply by Tantacrul** [reply]
For us the question is more about replatforming to Qt: "Would we dedicate resources to rebuild this feature?".

I agree that right now (in Audacity 3), there's no advantage in removing. However, for the future, if ever there was a feature where I cannot see the value, it's Karaoke mode :D

I currently can see it as the kind of thing that would be dropped and missed by virtually no one.

**Reply by Tantacrul** [reply]
(I think I'd need to do a bit of research to confirm we're right about this, of course)

**Comment by rbdannenberg** [comment]
Ironically, the Karaoke view was created to show off automatic alignment of lyrics to music, implemented by Dominic as a student in a course on speech recognition. The fact that it was _not_ "tedious"  motivated the Karaoke view in the first place. (But to continue the history lesson, dragging the Sphinx speech recognition system into Audacity was never done, nor would that have been a good idea, so lyric alignment _is_ manual and tedious. This is a good case for a module or plug-in.)

**Reply by petersampsonaudacity** [reply]
> This is a good case for a module or plug-in

I recall @Paul-Licameli proposing that he could/should do just that - makes sense to me.

_And @rbdannenberg - interesting history lesson Roger_ :-)

**Comment by hugofloresgarcia** [comment]
The (hopefully coming soon) Deep Learning module supports models that can generate a label track, given an input audio track. I can imagine a source separation -> speech-to-text pipeline, where the Deep Learning Effect is used to separate the vocals from the mix, and the Deep Learning Analyzer takes the isolated vocals and performs speech-to-text. This could pave the way for an auto-Karaoke :).

**Comment by petersampsonaudacity** [comment]
See IRL we do have Karaoke users - see this post from today on the Forum:
https://forum.audacityteam.org/viewtopic.php?f=70&t=123251

**Comment by Tantacrul** [comment]
The tough part is trying to determine the % if people who use it. As Leo pointed out earlier, perhaps the way to approach this (as we do with most other things) is to think of a solid replacement, rather than a complete deletion.

The unknown right now is how easily this can port to Qt. Ideally, we'd just keep it buried and replace it at some point in the future. If it requires lots of work to rebuild, then that is a tougher prospect.
